ramalama/huggingface.py

```diff
@@ -449,23 +470,25 @@ def _pull_with_model_store(self, args, debug: bool = False):
                 hf_repo = HuggingfaceRepository(name, organization, tag)
             snapshot_hash = hf_repo.model_hash
             files = hf_repo.get_file_list(cached_files)
-            self.store.new_snapshot(tag, snapshot_hash, files)
+            try:
+                self.store.new_snapshot(tag, snapshot_hash, files)
+            except Exception as e:
+                # Cleanup failed snapshot
+                try:
+                    self.store.remove_snapshot(tag)
+                except Exception as exc:
+                    if args.debug:
+                        perror(f"ignoring failure to remove snapshot: {exc}")
+                    # ignore any error when removing snapshot
+                    pass
+                raise e
         except Exception as e:
             if not self.hf_cli_available:
```

Comment:
@ericcurtin I think the huggingface-cli fallback logic could now be dropped but not 100% sure if there are other use-cases that it covers so leaving it in place for now...

Reply:
Which do you consider the fallback logic? :)

I was never a big fan of the hf cli compatibilty, I think we should do the hf client ourselves and store our versions of all the files somewhere here:

~/.local/share/ramalama/

kinda like what llama.cpp does.

Reply:
> Which do you consider the fallback logic? :)

This block.. catches all exceptions and tries again using huggingface-cli (if available). I suspect it was added to support pulling by repo.vs model but now that is implemented so suspect this can be dropped. Just wondering if its still required for something else.... Pulling safetensor models maybe?

Reply:
I think what you did makes sense, we should do our own huggingface client :)

Reply:
...and maybe clean-up the "old" model store code?!